primitives/arithmetic/src/traits.rs

```diff
@@ -728,14 +725,13 @@ mod ensure {
 		/// assert_eq!(underflow(), Err(ArithmeticError::Underflow));
 		/// ```
 		fn ensure_from(other: T) -> Result<Self, ArithmeticError> {
-			Self::try_from(other).map_err(|_| error::equivalent(other))
+			let err = error::equivalent(&other);
```

Comment:
Regarding the performance of this, it seems like the assembly resulting code of this can be easily reordered by the compiler and should not be penalty

Reply:
Did you check this?

Reply:
Not empirically (not sure how to do it well). 

To extend my argument, `error::equivalent` has no side effects, and if the resulting value is only read under a condition, the compiler can only execute that for that condition. I understand that this works for any T that is not really consumed by the `try_from()`, which means that T is in fact a `Copy` type. The compiler could not optimize this if T is not `Copy`.

Reply:
Okay ;)